### PR TITLE
Ensure that version used my plugin loader has the correct syntax (#4417)

### PR DESCRIPTION
**Pull request by sredxny**
* ensure that version used my plugin loader has the correct syntax

## Description

backport of https://github.com/TykTechnologies/tyk/pull/4417

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit:  
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3551224642)

**Review by tbuchaillot — APPROVED** [review]

**Review by mativm02 — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit: ca6c10055ed85f426d80da6ac880202508835b36 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3565847742)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit: c6d82a2399fc59b649033d35af6ce270af15184c 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3565922989)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit: 4f2ca2cfd1fc57461e0fb9f2b9e864192b92ddd0 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3566286394)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit: dc3b6897514b977a6cd1033380a0556bd115deca 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3566567543)

File: gateway/mw_go_plugin.go [hunk 1]
```diff
@@ -236,10 +246,10 @@ func (m *GoPluginMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Reque
 	return
 }
 
-// goPluginFromTykVersion builds a name of plugin based on tyk version
+// getGoPluginNameFromTykVersion builds a name of plugin based on tyk version
 // os and architecture. The structure of the plugin name looks like:
 // {plugin-dir}/{plugin-name}_{GW-version}_{OS}_{arch}.so
-func (m *GoPluginMiddleware) goPluginFromTykVersion(version string) string {
+func (m *GoPluginMiddleware) getGoPluginNameFromTykVersion(version string) string {
```
Comment (tbuchaillot) [inline gateway/mw_go_plugin.go:252]: @sredxny why are you adding Go to the func name? it's already a function for `GoPluginMiddleware`

Comment (sredxny) [reply gateway/mw_go_plugin.go:252]: ahhhh makes sense!

File: gateway/version.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
 package gateway
 
-var VERSION = "v4.1.0"
+var VERSION = "v4.3.0"
```
[the remark this answers is not in the document]
Comment (sredxny) [reply gateway/version.go:3]: yes

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit: 6f142c51b4218c5f77f4e6c936ef333017f27146 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3566761114)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4427/merge
Commit: 09ac312231f9c65a31ab208e1f1123451c8709b4 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3566837317)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/release-4.3
Commit: 377c6b74a22bf9e707227d52d660fc98291a710b Ensure that version used my plugin loader has the correct syntax (#4417) (#4427)


* ensure that version used my plugin loader has the correct syntax

* allow goplugin loader to load plugin without the v prefix in version

Co-authored-by: maciej <39672152+maciejwojciechowski@users.noreply.github.com>
Co-authored-by: Furkan Senharputlu <furkan_senharputlu@hotmail.com>
Triggered by: push (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3567204923)

**Pull request merged**